lib/local-storage-adapter.js

```diff
@@ -13,6 +14,12 @@ function createLocalStorageAdapter (queueName) {
   var backoffTimeKey = BACKOFF_TIME_KEY.replace('*', queueName)
   var errorCountKey = ERROR_COUNT_KEY.replace('*', queueName)
 
+  var cacheInSync = false
+  var storageInSync = true
+  var queueCache = []
+
+  var flushOnDefer = _.debounce(flush, 0)
```

Comment:
Most of the time all the events are emitted before Jolt actually starts so we should catch most cases. I may relax this a bit in the future to improve performance when that isn't the case but I figured this was the safest option for now (I want to make sure we drop as little data as possible)